packages/gatsby/cache-dir/static-query.js

```diff
@@ -2,9 +2,42 @@ import React from "react"
 import PropTypes from "prop-types"
 
 const StaticQueryContext = React.createContext({})
-let StaticQueryServerContext = null
+let _StaticQueryServerContext = null
 if (React.createServerContext) {
-  StaticQueryServerContext = React.createServerContext(`StaticQuery`, {})
+  _StaticQueryServerContext = createServerContext(`StaticQuery`, {})
+}
+
+function createServerOrClientContext(name, defaultValue) {
+  if (React.createServerContext) return createServerContext(name, defaultValue)
+  else return React.createContext(defaultValue)
+}
+
+// Ensure serverContext is not created more than once as React will throw when creating it more than once
+// https://github.com/facebook/react/blob/dd2d6522754f52c70d02c51db25eb7cbd5d1c8eb/packages/react/src/ReactServerContext.js#L101
+function createServerContext(name, defaultValue) {
+  if (!global.__GATSBY_SERVER_CONTEXT__) {
```

Comment:
`global` here refers to variables defined in the webpack config, in a browser context this will be a reference error. You likely want `globalThis`, see https://github.com/gatsbyjs/reach-router/pull/4/commits/9cd4453f5d6ea8180498bc51baed884a0cc0c938